src/test/suite/modes.test.ts

```diff
@@ -52,46 +51,46 @@ suite("Roo Code Modes", () => {
 				assert.fail("No messages received")
 			}
 
-			assert.ok(
-				globalThis.provider.messages.some(
-					({ type, text }) => type === "say" && text?.includes(`"request":"[switch_mode to 'code' because:`),
-				),
-				"Did not receive expected response containing 'Roo wants to switch to code mode'",
-			)
-			assert.ok(
-				globalThis.provider.messages.some(
-					({ type, text }) => type === "say" && text?.includes("software engineer"),
-				),
-				"Did not receive expected response containing 'I am Roo in Code mode, specializing in software engineering'",
+			await globalThis.provider.updateGlobalState("mode", "Ask")
```

Comment:
I think it might be better to use handleModeSwitch so it does the associated api config switch etc (in case we wanted to use another model to evaluate this someday)